### PR TITLE
Add version to upstream titles

### PR DESCRIPTION
**Pull request by lzap**
We have a branch support online now:

https://docs.theforeman.org/2.3/Planning_Guide/index-foreman.html

However it's not clear what guide version are you looking at. This introduces version number upstream to all titles, downstream there is a subtitle inserted but we don't do this workflow upstream. This will do it.

### CODE REVIEW

**Comment by lzap** [comment]
The PDF version looks a bit ugly, I think maybe I can put the version below as a new title or subtitle?

image

**Comment by lzap** [comment]
There appears to be some meta information which renders to the header and footer, but I can't get `{ProjectVersion}` variable working there. Weird. https://docs.asciidoctor.org/asciidoc/latest/document/header/

File: guides/common/attributes.adoc
Comment (ekohl) [inline guides/common/attributes.adoc:2]: This is the nightly version so shouldn't this be 2.4 and Katello 4.0?

Comment (lzap) [reply guides/common/attributes.adoc:2]: Yes I will add a new commit once we agree, I want to merge this into 2.3 branch.

**Comment by lzap** [comment]
Ok I figured out how to put metadata into the AsciiDoc header, now it renders nice both for HTML and PDF. It even puts version into footer in the end of the HTML version:

image

**Comment by lzap** [comment]
I rebased, let me know what you think about "Version X.Y, published YEAR". I wanted to signal clearly to the reader that it might be quite old version if they happen to randomly land on an older version from Google.

File: guides/common/attributes.adoc
Comment (ekohl) [reply guides/common/attributes.adoc:3]: FYI: nightly is about to bump to 2.5. Would you also branch 2.4?

Comment (melcorr) [reply guides/common/attributes.adoc:3]: From what I understand, Katello releases come slightly after a Foreman release. Is it possible to have a branch, for example, for Foreman 2.4 & Katello 3.18 AND Foreman 2.4 and Katello 4?

Comment (ekohl) [reply guides/common/attributes.adoc:3]: No, we only had 1 example where there was Katello 3.9 and Katello 3.10 on a single Foreman release but I don't really see that happening again. The biggest difference between 3.9 and 3.10 was EL8 support because it missed the 3.9 deadline. Overall it was a painful experience. I certainly hope we can avoid it.

Comment (melcorr) [reply guides/common/attributes.adoc:3]: Ok, even better. Less complications. Thank you @ekohl

**Comment by lzap** [comment]
So? What do you guys say? Shall I add a new commit on top of this one with the latest versions and shall we merge this? I'd like to backport this into 2.3 so it's online.

**Comment by melcorr** [comment]
If @ekohl is Ok with this, it LGTM.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc
Comment (ekohl) [inline guides/doc-Administering_Red_Hat_Satellite/master.adoc:9]: Is a year sufficient? Perhaps it needs to be a date?

**Review by ekohl — APPROVED** [review]
There does appear to be a lot of duplication but overall I think this makes sense.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc
Comment (lzap) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:9]: Date is at the end of each page (PDF version does not have it). I can add month but I would like to avoid full date.

In the github action, I actually `sed` date in the format of YYYY-MM-DD HH:MM to simply YYYY-MM because I don't think readers need to know exact date. It creates a change every time we commit into `gh-pages` and I would like to have sane history.

Alternatively, I can introduce some "nightly" variable which will make this to only appear on released branches. As long as we won't forget to flip it that could be the best solution.

Comment (lzap) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:9]: That variable might be actually a good idea - we could leverage it with values "nightly, stable, old" and utilize it to render some warnings for old and nightly content.

Comment (ekohl) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:9]: I think it makes sense to have smaller changes, but as a reader it implies it's only published yearly. At least a PDF gives that impression.

In our current manual we also have something like "this isn't the latest version, see $stable" and "this is nightly, see $stable for a supported version". Having a similar thing here does make sense to me. Perhaps even something more complex: note when it's supported. Currently Foreman 2.2 and 2.3 are supported, but 2.1 and earlier can get a big warning that the release is end of life. Especially for an update guide that's great to have. Nobody should install an EOL release unless they know what they're doing.

**Comment by lzap** [comment]
Ok here is another version. Much more clean now.

When built as "foreman" or "foreman-deb" with "nightly" attribute the header reads `Version Nightly, published Feb 2021` and the footer reads `Version Nightly` and on the second line `Last updated Feb 2021`. With "stable" attribute it is the same except there is a version instead "Nightly" and the date is `Jan 15 2021`. And unsupported is the same except version reads `Version 2.1 (unsupported)`. When built as "satellite" nothing is added, versioning is done differently downstream.

File: .github/workflows/deploy.yml
Comment (lzap) [inline .github/workflows/deploy.yml:65]: This hack is no longer needed since `Last updated` is now rendered correctly due to `-a docdatetime="$(DATE_MY)"` `Makefile` change.

File: guides/doc-Administering_with_Ansible/master.adoc
Comment (ekohl) [inline guides/doc-Administering_with_Ansible/master.adoc:19]: Is there a way to avoid this duplication everywhere? Perhaps via an include?

**Comment by lzap** [comment]
Note to myself: Edit Github Action and modify this line if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/2.3' to include the new version. Only last two stable versions are needed.

This should be done in the branch. Maybe we can find a generic way, regexp like X.Y @ekohl if you have an idea.

File: guides/doc-Administering_with_Ansible/master.adoc
Comment (lzap) [reply guides/doc-Administering_with_Ansible/master.adoc]: How I could miss this! Of course.

**Comment by lzap** [comment]
Ok @ekohl helped me to figure out better solution, no need to update GH Action if it works. Let's see.

**Comment by lzap** [comment]
All green, I am unable to test if deployment now works so let's just merge this if there are no objections and I will fix it in the branch if needed.

**Pull request merged**
